### ISSUE TITLE
Software Updates: Redesign details

### ISSUE THREAD

**Issue by martinpitt** [cockpit-project/cockpit#8379]
@andreasn made a new design for the software updates list:

![design](https://raw.githubusercontent.com/cockpit-project/cockpit-design/master/software-updates/software-updates-security-levels.png)

For comparison, current design: https://youtu.be/BENWI-dTKMs

I have some questions/issues with this which need to be resolved before I can implement this:

 1. "Advisory Type" only makes sense on RHEL; can we find a generic label to avoid different strings on different OSes (which are a nuisance for i18n)? Like "Type", "Issues", "References"
 2. Please confirm: Always only show the first line of description in "Details", and the full one in the "Details" tab of the expander? This would remove the default restriction to 15 lines and the "More..." link?
 3. What does the cog (or screw head?) vs. magic wand icons mean in "Type"? I suppose they should be the same, and that the icon definition changed between the two drafts? I take it the three numbers/icons are "security", "bug fix", and "enhancement"?
 4. Where does the bugs list go now? Note that usually we just have an URL and just show a number, but don't know anything else about these (we don't want to automatically fetch these URLs, it's a privacy issue and slow). My gut feeling is that this bug link list should just go into "Details", below the description.
 5. "Errata" only makes sense in the context of RHEL. This tab in the expander would just not be shown on other OSes, like in "Fedora".
 6. Errata: common case is that there is only one; there might be several when not installing updates for a longer time; but usually, an erratum might contain several packages, but one single package update version always just has one erratum.
 7. If there are no Errata, should the expander still have a single "Details" tab, or hide the tab line? (I assume the latter). This also depends on where to show the bugs.
 8. Errata: We do have the "issued" time in the updateinfo, but it is not exposed through PackageKit.  There is not a good locally available string for the "Synopsis" column. It would need to be fetched from the URL, but again this is slow and also a potential privacy problem. Also, the urgency is per-package, not per-advisory; so showing different urgencies in the top-level table is fine, but not in the different advisories if there are multiple ones.

This is the structured metadata that we have (synthetic info, but structure taken from a real-life update), aside from the textual description:
```
Details about the update:
 Package: foo_1.2_noarch
 Vendor: https://access.redhat.com/errata/RHSA-2018:0001, RHSA-2018:0001, https://access.redhat.com/security/updates/classification/#important, important,  https://access.redhat.com/security/cve/CVE-2017-0000
 Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1234567
 Restart: system
 Update text: Package description of foo
```

**Comment by martinpitt** [comment]
Notes to self: the different security levels can and should be implemented after the general re-design. Technically, security levels and references to errata are implemented in the yum _updateinfo xml like this:
```xml
<reference href="https://access.redhat.com/errata/RHSA-2018:0029" id="RHSA-2018:0029" title="RHSA-2018:0029" type="self" />
<reference href="https://access.redhat.com/security/updates/classification/#moderate" id="" title="" type="other" />
```
So we have a way to mock this in tests. @andreasn: Can you please re-confirm the shield icons in the design? https://access.redhat.com/security/updates/classification uses different styles of icons; same colors, but they are hollow.

**Comment by martinpitt** [comment]
Results from meeting:
1. "Type" is fine for now.
2. Confirm: Always show the first line only; in the expander, limit to 15 lines and make overflow scrollable
3. magic wand is obsolete; always the cog; it's for enhancement
4. bug list goes under (mobile) or to the right of (desktop) the changelog
5. confirmed
6. PK bug: expose erratum issue date through the PackageKit API
7. obsolete, see 8
8. show errata as link list similar to bugs, in Details tab

**Comment by martinpitt** [comment]
I have a first iteration on the new design, with the changes from the previous comments. I did this with pkg/packagekit/mock-updates.es6 (see comment at the top of the file how to enable it).

This is the initial view with everything collapsed:

![updates-collapsed](https://user-images.githubusercontent.com/200109/34827916-07ad9486-f6dc-11e7-8560-97f5709a828c.png)

I. e. security-one has two CVEs, security-two has one, buggy has 25 bug fixes; manypkgs and verbose don't have bug or CVE links, they just show the reported severity from yum data. This demonstrates the truncation of long package name lists and descriptions.

When expanding all but security-two one sees the details. This is 80% zoom in order to fit my screen; sorry that it's a bit harder to read. This illustrates the scrolling of overly long changelogs, and that bug/CVE/package lists are complete here and wrap accordingly:

![updates-expanded](https://user-images.githubusercontent.com/200109/34827919-08fdf448-f6dc-11e7-97f9-50e430938070.png)

I also tested mobile view, looks reasonable.

I still need to adjust our tests, so no PR yet, but I'd appreciate some initial feedback.

Note that there are no "security levels" yet, this will be a change building up on top of this. Also, the icons are still wrong as our PatternFly version doesn't yet have the newly proposed ones. I'll update PF in a separate PR.
